docs/en/tutorials/getting-started.rst

```diff
@@ -228,33 +227,196 @@ entity definition:
+            {
+                $this->passwordHash = $hash($password);
+            }
+
+            public function ban(\DateInterval $duration): void
+            {
+                assert($duration->invert !== 1);
+
+                $this->bans[] = new Ban($this);
+            }
         }
 
-        public function setName($name)
+Additionally, our entities should never see their state change
+partially without validation. For example, creating a ``new Product()``
+object without any data makes it an **invalid object**.
+Rich models should represent **behavior**, not **data**, therefore
+they should be valid even after a ``__construct()``.
+
+To help creating such objects, we can rely on ``DTO``s, and/or make
+our entities always up-to-date. This can be performed with static constructors,
+or rich mutators that accept the ``DTO`` as argument. And DTOs can be
+validated before being sent to the object for update.
+
+By using DTOs, if we take our previous ``User`` example, we could create
+a ``ProfileFormUser`` DTO object that will be a plain model, totally
+unrelated to our database, that will be populated via a form and validated.
+Then we can add a new mutator to our ``User``:
+
+.. configuration-block::
+
+    .. code-block:: php
+        <?php
+        class User
         {
-            $this->name = $name;
+            public function updateFromProfile(ProfileFormUser $profileForm): void
+            {
+                // ...
+            }
+
+            public static function createFromRegistration(RegistrationFormUser $registrationForm): self
+            {
+                // ...
+            }
         }
-    }
-
-When creating entity classes, all of the fields should be protected or private
-(not public), with getter and setter methods for each one (except $id).
-The use of mutators allows Doctrine to hook into calls which
-manipulate the entities in ways that it could not if you just
-directly set the values with ``entity#field = foo;``
 
-The id field has no setter since, generally speaking, your code
-should not set this value since it represents a database id value.
-(Note that Doctrine itself can still set the value using the
-Reflection API instead of a defined setter function)
+There are several advantages to using such model:
+
+* Entity **state is always valid**. No setters means that we only
+update portions of the entity that should already be valid.
+* Instead of having plain getters and setters, our entity now has
+**real behaviors**, meaning it is much easier to determine the
+logic in the domain.
+* DTOs can be reused in other components, like when deserializing
+mixed content, using forms...
+* Classic and static constructors can be used to manage different
+ways to create our objects, and they can also use DTOs.
+* Anemic models tend to isolate the entity from logic, whereas
+rich models fix that because logic related to the entity can be
```

Comment:
>  fix that because

"fix that" is strange to read in this context, because it is not immediately obvious that this is a problem. We only briefly explained that anemic models lead to some issues, but we didn't explain that logic outside entities versus inside is worse.